src/world/conquest_system.cpp

```diff
@@ -20,3 +20,342 @@ along with this program.  If not, see http://www.gnu.org/licenses/
+
+    // 3- Create ZMQ Message and queue it
+    zmq::message_t dataMsg = zmq::message_t(dataLen);
+    memcpy(dataMsg.data(), data, dataLen);
+
+    if (ipp == 0)
+    {
+        queue_message_broadcast(MSG_WORLD2MAP_REGIONAL_EVENT, &dataMsg);
+    }
+    else
+    {
+        queue_message(ipp, MSG_WORLD2MAP_REGIONAL_EVENT, &dataMsg);
+    }
+}
+
+bool ConquestSystem::updateInfluencePoints(int points, unsigned int nation, REGION_TYPE region)
+{
+    if (region == REGION_TYPE::UNKNOWN)
+    {
+        return false;
+    }
+
+    std::string Query = "SELECT sandoria_influence, bastok_influence, windurst_influence, beastmen_influence FROM conquest_system WHERE region_id = %d;";
+
+    int ret = zmqSql->Query(Query.c_str(), static_cast<uint8>(region));
+
+    if (ret == SQL_ERROR || zmqSql->NextRow() != SQL_SUCCESS)
+    {
+        return false;
+    }
+
+    int influences[4] = {
+        zmqSql->GetIntData(0),
+        zmqSql->GetIntData(1),
+        zmqSql->GetIntData(2),
+        zmqSql->GetIntData(3),
+    };
+
+    if (influences[nation] == 5000)
+    {
+        return false;
+    }
+
+    auto lost = 0;
+    for (auto i = 0u; i < 4; ++i)
+    {
+        if (i == nation)
+        {
+            continue;
+        }
+
+        auto loss = std::min<int>(points * influences[i] / (5000 - influences[nation]), influences[i]);
+        influences[i] -= loss;
+        lost += loss;
+    }
+
+    influences[nation] += lost;
+
+    ret = zmqSql->Query("UPDATE conquest_system SET sandoria_influence = %d, bastok_influence = %d, "
+                        "windurst_influence = %d, beastmen_influence = %d WHERE region_id = %u;",
+                        influences[0], influences[1], influences[2], influences[3], static_cast<uint8>(region));
+
+    return ret != SQL_ERROR;
+}
+
+void ConquestSystem::updateWeekConquest()
+{
+    TracyZoneScoped;
+
+    // 1- Notify all zones that tally started
+    sendTallyStartMsg();
+
+    // 2- Do the actual db update
+    const char* Query = "UPDATE conquest_system SET region_control = \
+                            IF(sandoria_influence > bastok_influence AND sandoria_influence > windurst_influence AND \
+                            sandoria_influence > beastmen_influence, 0, \
+                            IF(bastok_influence > sandoria_influence AND bastok_influence > windurst_influence AND \
+                            bastok_influence > beastmen_influence, 1, \
+                            IF(windurst_influence > bastok_influence AND windurst_influence > sandoria_influence AND \
+                            windurst_influence > beastmen_influence, 2, 3)));";
+
+    int ret = zmqSql->Query(Query);
+    if (ret == SQL_ERROR)
+    {
+        ShowError("handleWeeklyUpdate() failed");
+    }
+
+    // 3- Send tally end Msg
+    sendRegionControlsMsg(CONQUEST_WORLD2MAP_WEEKLY_UPDATE_END);
+}
+
+void ConquestSystem::updateHourlyConquest()
+{
+    sendInfluencesMsg(true);
+}
+
+void ConquestSystem::updateDailyTallyPoints()
```

Comment:
Not sure if this is the best place for this since it has no relation to the conquest system.

Reply:
Moved it back. I was confused by the word tally. Thanks!